packages/wrangler/src/__tests__/versions/secrets/put.test.ts

```diff
@@ -0,0 +1,228 @@
+import { describe, expect, test } from "vitest";
+import { mockAccountId, mockApiToken } from "../../helpers/mock-account-id";
+import { mockConsoleMethods } from "../../helpers/mock-console";
+import { clearDialogs, mockPrompt } from "../../helpers/mock-dialogs";
+import { useMockIsTTY } from "../../helpers/mock-istty";
+import { useMockStdin } from "../../helpers/mock-stdin";
+import { runInTempDir } from "../../helpers/run-in-tmp";
+import { runWrangler } from "../../helpers/run-wrangler";
+import writeWranglerToml from "../../helpers/write-wrangler-toml";
+import { mockPostVersion, mockSetupApiCalls } from "./utils";
+
+describe("versions secret put", () => {
+	const std = mockConsoleMethods();
+	const { setIsTTY } = useMockIsTTY();
+	runInTempDir();
+	mockAccountId();
+	mockApiToken();
+	afterEach(() => {
+		clearDialogs();
+	});
+
+	test("can add a new secret (interactive)", async () => {
+		setIsTTY(true);
+
+		mockPrompt({
+			text: "Enter a secret value:",
+			options: { isSecret: true },
+			result: "the-secret",
+		});
+
+		mockSetupApiCalls();
+		mockPostVersion((metadata) => {
+			expect(metadata.bindings).toStrictEqual([
+				{ type: "secret_text", name: "NEW_SECRET", text: "the-secret" },
+			]);
+			expect(metadata.keep_bindings).toStrictEqual([
+				"secret_key",
+				"secret_text",
+			]);
+		});
+		await runWrangler(
+			"versions secret put NEW_SECRET --name script-name --x-versions"
+		);
+
+		expect(std.out).toMatchInlineSnapshot(`
+			"🌀 Creating the secret for the Worker \\"script-name\\"
+			✨ Success! Created version id with secret NEW_SECRET.
+			➡️  To deploy this version with secret NEW_SECRET to production traffic use the command \\"wrangler versions deploy --x-versions\\"."
+		`);
+		expect(std.err).toMatchInlineSnapshot(`""`);
+	});
+
+	// For some reason, this always hangs. Not sure why
+	test.skip("can add a new secret (non-interactive)", async () => {
```

Comment:
not sure if you have time to fix this test today, but if not, let's pls test manually that this works...if you haven't done that already...and maybe take fixing the test as a afast-follow?

Reply:
yeah this has been tested manually, fixing will be a fast follow
I think some prompt or something isn't being mocked fully - i just wish i knew which haha